src/Resources/AbstractResource.php

```diff
@@ -8,11 +8,34 @@ abstract class AbstractResource
 {
     public function toArray(): array
     {
-        return (array) $this;
+        $array = [];
+
+        $this->__objectToArray($this, $array);
+
+        return $array;
     }
 
     public function toJson(): string
     {
         return json_encode($this);
     }
+
+    private function __objectToArray($obj, &$arr)
+    {
+        if (! is_object($obj) && ! is_array($obj)) {
+            $arr = $obj;
+
+            return $arr;
+        }
+        foreach ($obj as $key => $value) {
+            if (! empty($value)) {
+                $arr[$key] = [];
+                $this->__objectToArray($value, $arr[$key]);
```

Comment:
oh, now I see. This method could use some renaming. I'm also not sure we'd want to weed out empty values, as things like `0` could be extremely valid.

Reply:
Oh, good catch. That `0` thing could have been a problem.

(this method was copied from chatGPT/SO for recursively converting the object(s) into an array)

Reply:
Actually, it's not dropping any values. If it's empty it just sets the key to the value. I ran some tests and it appears to be working fine.

Only thing I'm noticing now is that for the responses that are arrays of objects (like paginated products response) you have to loop through the array and then do `->toArray()` if you want everything as arrays.

Unrelated here but I wonder if I need a collection object so then you can always use `->toArray()` on the response data. Thoughts?

Reply:
Okay, although it still doesn't seem to make a ton of sense (if it's empty, set the value to the key, if not, create an empty array for the key, then run this method on the value and empty array just added).

Whether it technically works or not (and would it for empty arrays?) this is really not code from ChatGPT. Compare it to the 5-line example I provided.

For the other stuff, I dunno, I'm losing the thread here.